Add package: khard

matthiasbeyer opened the pull request:
Had to wait until vdirsyncer was merged.

Will test ASAP.

matthiasbeyer committed:
khard: init at 0.4.0

matthiasbeyer commented:
Can someone explain why there is ni `stdenv` available for this package (see build failure message)?

pSub commented:
Because you don't pass it as an argument.

Change `{ fetchurl, pythonPackages }:` into `{ stdenv, fetchurl, pythonPackages }:`

matthiasbeyer commented:
Oh, this is embarrassing! I thought I have it...

jagajaga commented:
Please, don't make lots of commits in PR. When you fix something just squash commits and do git --force push.

pSub commented:
It is also a good idea to mention that something is work-in-progress in the title of the PR.

matthiasbeyer commented:
Can someone tell me what went wrong here and how to fix the bug?

jagajaga commented:
First of all - do not make lot's of commits, as I mention before.

And it can't find `xslt-config` as I understand.

matthiasbeyer commented:
@jagajaga : How can I fix this?

And what do you mean by "making lots of commits"? They are supposed to be squashed!

jagajaga commented:
Find and add this pkg to deps :)

Now I see that there are 4 commits and 3 of them need to be squashed into 1st one. So yes, they are supposed to be squashed, but they are not :)

jagajaga commented:
This can help you http://stackoverflow.com/questions/5178416/pip-install-lxml-error?

matthiasbeyer commented:
From what I see on my local machine, it cannot find `argparse`! Which is strange, because I actually have specified argparse from the `pythonPackages` set...

jagajaga commented:
Any updates? :)

matthiasbeyer commented:
Not yet, I have not enough time atm, but I will have a closer look at the weekend.

matthiasbeyer commented:
hmm... I'm still getting the backtrace that `argparse` is not found (updated to latest master, which is e30f14b791053a1d8634b0ba049a5e9d49f76452 atm):

```
patching script interpreter paths in /nix/store/fam9xzqsf2pc5y0wriwmk7j9z7az5x74-vdirsyncer-0.4.3
/nix/store/fam9xzqsf2pc5y0wriwmk7j9z7az5x74-vdirsyncer-0.4.3/bin/vdirsyncer
wrapping `/nix/store/fam9xzqsf2pc5y0wriwmk7j9z7az5x74-vdirsyncer-0.4.3/bin/vdirsyncer'...
building path(s) ‘/nix/store/qkpr2vafnrkd8ic7x8lbikwnsivn98w5-python2.7-khard-0.2.1’
unpacking sources
unpacking source archive /nix/store/c7dlmxjgrf12gixfb961qqy8nxmzlm6v-v0.2.1.tar.gz
source root is khard-0.2.1
patching sources
configuring
building
running build
running build_py
creating build
creating build/lib
creating build/lib/khard
copying khard/khard.py -> build/lib/khard
copying khard/helpers.py -> build/lib/khard
copying khard/config.py -> build/lib/khard
copying khard/carddav_object.py -> build/lib/khard
copying khard/__init__.py -> build/lib/khard
creating build/lib/davcontroller
copying davcontroller/davcontroller.py -> build/lib/davcontroller
copying davcontroller/__init__.py -> build/lib/davcontroller
running tests
running test
Searching for argparse
Reading https://pypi.python.org/simple/argparse/
Download error on https://pypi.python.org/simple/argparse/: [Errno -2] Name or service not known -- Some packages may not be found!
Couldn't find index page for 'argparse' (maybe misspelled?)
Scanning index of all packages (this may take a while)
Reading https://pypi.python.org/simple/
Download error on https://pypi.python.org/simple/: [Errno -2] Name or service not known -- Some packages may not be found!
No local packages or download links found for argparse
error: Could not find suitable distribution for Requirement.parse('argparse')
builder for ‘/nix/store/h3vwximp7kqpaavbgacwxpbv4xnmdxna-python2.7-khard-0.2.1.drv’ failed with exit code 1
error: build of ‘/nix/store/h3vwximp7kqpaavbgacwxpbv4xnmdxna-python2.7-khard-0.2.1.drv’ failed
```

(previous installation backtrace ommitted)

matthiasbeyer commented:
Updated onto latest master and added fixup commit to update to khard version 0.3.3. The mentioned missing argparse error is still present, though.

Any pointers?

domenkozar commented on pkgs/top-level/all-packages.nix:
```diff
@@ -11394,6 +11394,12 @@ let
+  khard = callPackage ../applications/misc/khard {
+    # does not work with python3Packages because vobject does not work with
+    # python3Packages
+    pythonPackages = pythonPackages;
```
No need to pass that in as it's the same as default

domenkozar commented on pkgs/applications/misc/khard/default.nix:
```diff
@@ -0,0 +1,25 @@
+    configobj
+    vobject
+    argparse
+    pkgs.vdirsyncer
```
does vdirsyncer really need to be propagated

matthiasbeyer replied:
I'm not sure what "propagated" means in this context.

domenkozar replied:
It means dependency is propagated to any packages that depend on khard

matthiasbeyer replied:
Oh. Then no, I guess not.

jagajaga replied:
@matthiasbeyer what about this discussion?

matthiasbeyer replied:
Ah, forgot about this one. I must add the `vdirsyncer` package to the `buildDepends` list instead of the `propagatedBuildInputs` list, right?

domenkozar commented:
Is this a cli tool? If so, it needs to have `namePrefix = "";`

matthiasbeyer commented:
Waiting for the patches in #7872 , which partly fixes this package.

matthiasbeyer commented:
I test-build this package and it works for me.

Thanks for your patience!

matthiasbeyer commented:
Updated this, works for me now. Also this adds `0.4.0` now, not `0.3.3` anymore...
